model.py

```diff
@@ -1,103 +1,1222 @@
+        # significantly change the location of the changepoints.
+        if (dphi_change > 1.2) or (q_change > 1.2) or (rwd_change > 1.2):
+            # Calculate inclination
+            inc = roche.findi(q.currVal, dphi.currVal)
+
+            # Calculate wd contact phases 3 and 4
+            phi3, phi4 = roche.wdphases(q.currVal, inc, rwd.currVal, ntheta=10)
+
+            # Calculate length of wd egress
+            dpwd = phi4 - phi3
+
+            # Distance from changepoints to mideclipse
+            dist_cp = (dphi.currVal+dpwd)/2.
+
+            # save these values for speed
+            self._dist_cp = dist_cp
+            self._oldq = q.currVal
+            self._olddphi = dphi.currVal
+            self._oldrwd = rwd.currVal
+        else:
+            # Use the old values
+            dist_cp = self._dist_cp
+
+        # Find location of all changepoints
+        min_ecl = int(np.floor(self.lc.x.min()))
+        max_ecl = int(np.ceil(self.lc.x.max()))
+
+        eclipses = [e for e in range(min_ecl, max_ecl+1)
+                    if np.logical_and(e > self.lc.x.min(),
+                                      e < 1+self.lc.x.max()
+                                      )
+                    ]
+
+        changepoints = []
+        for e in eclipses:
+            # When did the last eclipse end?
+            egress = (e-1) + dist_cp
+            # When does this eclipse start?
+            ingress = e - dist_cp
+            changepoints.append([egress, ingress])
+
+        return changepoints
+
+    def createGP(self):
+        """Constructs a kernel, which is used to create Gaussian processes.
+
+        Using values for the two hyperparameters (amp,tau), amp_ratio and dphi,
+        this function: creates kernels for both inside and out of eclipse,
+        works out the location of any changepoints present, constructs a single
+         (mixed) kernel and uses this kernel to create GPs"""
+
+        # Get objects for ampin_gp, ampout_gp, tau_gp and find the exponential
+        # of their current values
+        ln_ampin = self.ampin
+        ln_ampout = self.ampout
+        ln_tau = self.tau
+
+        ampin = np.exp(ln_ampin.currVal)
+        ampout = np.exp(ln_ampout.currVal)
+        tau = np.exp(ln_tau.currVal)
+
+        # Calculate kernels for both out of and in eclipse WD eclipse
+        # Kernel inside of WD has smaller amplitude than that of outside
+        # eclipse.
+
+        # First, get the changepoints
+        changepoints = self.calcChangepoints()
+
+        # We need to make a fairly complex kernel.
+        # Global flicker
+        kernel = ampin * george.kernels.Matern32Kernel(tau)
+        # inter-eclipse flicker
+        for gap in changepoints:
+            kernel += ampout * george.kernels.Matern32Kernel(tau, block=gap)
+
+        # Use that kernel to make a GP object
+        georgeGP = george.GP(kernel, solver=george.HODLRSolver)
+
+        return georgeGP
+
+    def calc(self, plot=False):
```

Comment:
Same issue as raised above, here you've got ```calc``` returning ```ln_like```  - it should return the results of evaluating the model, and it can be used within a seperate ```ln_like``` function

Reply:
The more I think about it, the more I think you're right. It's definitely confusing as is, this chunk is gonna get overhauled.